goss-testing/scripts/check_weave.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+set -exo pipefail
+
+TMPFILE=$(mktemp)
+
+WEAVE_POD=$(kubectl -n kube-system get pods -l name=weave-net -o jsonpath='{.items[*].metadata.name}' | awk '{ print $1 }')
```

Comment:
This only checks one pod.   To be sure, we should check all weave pods.

Reply:
Yeah, there were several things wrong with the original goss test.   It wasn't really testing anything because it wasn't running the correct command.   I see now that the test suites with that test are run on each master and worker.  We only need to run the check on the node or on the pod for that one node.    I was thrown off because the original goss test was always running it on the same pod so I thought that it was only running the test once (not once on each node).

Running `weave --local status connections` and `kubectl -n kube-system exec "$pod" -c weave -- /home/weave/weave --local status connections` are doing the same thing.  So you can just remove the part that is looking over the pods.

Reply:
Okay -- change has been made and tested.